src/kubernetes.cc

```diff
@@ -757,13 +761,23 @@ struct Watcher : public std::thread {
         LOG(ERROR) << name_ << " => Unlocking completion mutex";
       }
       service_->stop();
+    } else {
```

Comment:
why we will check `config_.KubernetesUpdaterWatchReconnectMaxSilenceSeconds() > 0` in another place, but not here?

Reply:
I've added a check.